bokehjs/src/lib/models/widgets/base_date_picker.ts

```diff
@@ -91,6 +96,22 @@ export abstract class BaseDatePickerView extends PickerBaseView {
     }
     return result
   }
+
+  protected _on_close(selected: Date[]): void {
+    switch (selected.length) {
+      case 0:
+      case 1: {
+        // Incomplete selection, treat as no selection.
+        this.model.value = null
+        break
+      }
+      case 2:
+        break
+      default: {
+        unreachable("invalid length")
+      }
+    }
+  }
```

Comment:
Looking closer, could the switch just be replaced entirely by `if (selected.length < 2)` ?